### DISCUSSION TITLE
Blazor support?

### DISCUSSION THREAD

**Discussion by nerocui** [taublast/DrawnUi#1] [Ideas]
Hi, awesome library! There's no reason this shouldn't work on Blazor right? Have you thought about unifying all platform (include web) with Blazor (and mobile binding) or Uno platform? That'd be my dream framework.

**Comment by taublast** [comment]
Hi, I would love this lib to work with the existing SkiaSharp for Blazor too! Actually gathering thoughts from different people about the best approach to take to keep its maximum uniformity between MAUI and Blazor. if you would like to share your ideas please ping! :)

**Comment by nerocui** [comment]
I would say to either go with Blazor or Uno. Both can work cross platform and supports web. 

With Blazor, you'd be targetting web with Blazor WebAssembly and other native platform with Blazor mobile binding. The downside tho is that Blazor mobile binding is still experimental and doesn't have all the platform API support like MAUI. 

With Uno, you can pretty much support anything. The catch is that you don't have Blazor's Ecosystem for interacting with the web. Like easy way to add css and js to your component. But those can be added pretty easily on either your part or from Uno itself.

**Comment by ToolmakerSteve** [comment]
While I like the Uno approach in principle, pragmatically Blazor seems the way to best leverage Microsoft's apparent direction. It also better fits the fact that web technologies are dominant and spreading to non-web use cases. [And my client has asked about adapting their mobile app to Blazor. His main programmer and I are doing all [2D and UI] graphics via SkiaSharp now; we abandoned Xamarin Forms after spending a year on it a while ago, and so far have decided not to switch to Maui; so this will be a great SkiaSharp + Blazor investigation.] I am about to start experimenting with Blazor, both in WebAsm and in mobile apps. I also just started looking at SkiaSharp in WebAsm. More info when I know more.

**Comment by ToolmakerSteve** [comment]
> Blazor mobile binding is still experimental and doesn't have all the platform API support like MAUI.

This is not a problem. Under .Net 8, in platform-specific apps (the equivalent of "Xamarin.Android" and "Xamarin.iOS" native apps), I use various "Maui Esssentials" packages, to support mobile platform features. Doing so only requires .Net 8; doesn't require any "Maui UI". I haven't personally made bindings between Razor components and the Essentials APIs, but doing so is reasonably straightforward.

If we switch my client's app to Blazor, on desktop and mobile devices, we would use the bare minimum of Maui's hosting, to create one window containing a Blazor UI overlay and a SkiaSharp-on-OpenGL overlay. I'll post a sample when I have it. In a month or two. Including examples of access to Essentials APIs.

NOTE: AFAIK, the experimental [MobileBlazorBindings](https://github.com/dotnet/MobileBlazorBindings) are moot now. Do this the official way via Maui Blazor. TBD how to avoid pulling in unneeded Maui dlls and classes, if only want the Blazor side of Maui. Deal with that after have a working example.

**Comment by tcsaddul** [comment]
There is an active fork here:  https://github.com/Dreamescaper/BlazorBindings.Maui

**Reply by taublast** [reply]
Thanks I'll try this! :)

**Comment by taublast** [comment]
Blazor version is being worked on and will be out this year. To be used with Fluent C# mainly.
